client.go

```diff
@@ -986,6 +1125,25 @@ func TLSConfig(config *tls.Config) ClientOption {
 	}
 }
 
+func SkipVersionCheck() ClientOption {
```

Comment:
That can be a bit problematic since the order of passed options may result in different outcomes.
Currently, the user has the only option to set `versionCheck==false`:
```
NewClient(uri) => versionCheck == true
NewClient(uri, SkipVersionCheck()) => versionCheck == false
NewClient(uri, SkipVersionCheck(), LegacyMode(...)) => versionCheck = false
NewClient(uri, LegacyMode(...), SkipVersionCheck()) => versionCheck = false
```
If we use `VersionCheck(bool)` instead, the user gains the ability to set `versionCheck==true`:
```
NewClient(uri) => versionCheck == true
NewClient(uri, VersionCheck(false)) => versionCheck == false
NewClient(uri, VersionCheck(true)) => versionCheck == true
NewClient(uri, VersionCheck(true), LegacyMode(...)) => versionCheck = false
NewClient(uri, LegacyMode(...), VersionCheck(true)) => versionCheck = true
```
The order of `VersionCheck` and `LegacyMode` changes what value `versionCheck` has, which is not expected. (The older `SkipVersionCheck(bool)` has he same issue)